src/bin/ion/commands/beta/generate/mod.rs

```diff
@@ -102,7 +101,14 @@ impl IonCliCommand for GenerateCommand {
         println!("Started generating code...");
 
         // generate code based on schema and programming language
-        CodeGenerator::new(language, output).generate(schema)?;
+        match language {
```

Comment:
(🗺️ PR tour) Since the new changes introduced `CodeGenerator<Language>`, the initiation of code generation process uses specific `generate` method based on programming language.